### PR TITLE
test: Save javascript logs on failure

### PR DESCRIPTION
**Pull request by larskarlitski**
These logs sometimes contain information relevant to a failure.

### CODE REVIEW

File: test/common/phantom-driver.js [hunk 1]
```diff
@@ -46,7 +46,8 @@ var page = require('webpage').create();
 var sys = require('system');
 var fs = require('fs');
 var clearedStorage = false;
-var messages = "";
+var messages = [];
+var firstMessageIndex = 0;
```
Comment (martinpitt) [inline test/common/phantom-driver.js]: The "first" is a bit confusing. I'm not entirely sure when `respond()` gets called (it's hopelessly overloaded even in this file alone), but maybe something like `currentCallMessageIndex`?

Comment (larskarlitski) [inline test/common/phantom-driver.js]: Makes much more sense. Changed it to `currentCallMessageIndex`. Thanks!

**Comment by martinpitt** [comment]
Hmm, a bit hard to get the tests green. But the failures do point out a wart: Can you please update the code to not write empty files? They seem like useless clutter.

**Comment by larskarlitski** [comment]
> Can you please update the code to not write empty files? They seem like useless clutter.

I initially decided against that, because I thought not having the file in some cases might confuse people.

Now that you bring it up though, I agree that since this file will be empty more often than not, it's often just clutter. Removed.

**Pull request merged**